pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -56,6 +60,12 @@ var (
 	emptyMap = make(map[string]string)
 )
 
+const (
+	maxReplacements    = 1
+	lockAcquireTimeout = 1 * time.Second
+	cacheUpdateTimeout = 1 * time.Second
```

Comment:
For now it was a guess work . In reality its around 2sec , because we give the reflector update further 1 sec in case this `cacheUpdateTimeout` happens in `waitForFailedMachineCacheSync`.
Open for suggestions